src/analysis/special_functions/pow.lean

```diff
@@ -1065,6 +1065,20 @@ begin
     { simp [coe_rpow_of_ne_zero h, h] } }
 end
 
+lemma rpow_eq_top_of_nonneg (x : ennreal) {p : ℝ} (hp0 : 0 ≤ p) : x ^ p = ⊤ → x = ⊤ :=
+begin
+  rw ennreal.rpow_eq_top_iff,
+  intro h, cases h,
```

Comment:
put on two lines: there should be one single command per line, unless you close a simple goal right away (like you do on next line).